builder/amazon/common/ssh.go

```diff
@@ -34,7 +34,7 @@ func SSHHost(e ec2Describer, private bool) func(multistep.StateBag) (string, err
 				} else if i.PrivateIpAddress != nil && *i.PrivateIpAddress != "" {
 					host = *i.PrivateIpAddress
 				}
-			} else if private {
+			} else if private && i.PrivateIpAddress != nil && *i.PrivateIpAddress != "" {
```

Comment:
@tarrall I introduced some guards here. They are (at least) needed for the test but I think we should not count on that the AWS API always return stuff in general.